oxen-rust/test-put-demo.sh

```
@@ -0,0 +1,477 @@
+    SERVER_PID=$!
+    echo "Server started with PID: $SERVER_PID"
+    echo "Waiting for server to be ready..."
+    sleep 5
+fi
+
+# Step 3: Create local repository
+echo
+echo "Step 3: Creating local repository..."
+rm -rf demo-repo
+mkdir demo-repo
+cd demo-repo
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)